pgoapi/poke_utils.py

```diff
@@ -1,11 +1,89 @@
+            "TCPMDif": float(row["TCPM Difference"])
+        }
+        tcmpVals.append(float(row["TotalCpMultiplier"]))
+
+def setMaxCP(pokemon, maxTCPM, game_master):
+    pokeGameData = game_master.get(pokemon.pokemon_id, PokemonData())
+    if int(pokeGameData.PkMn) == 0 or not all_in(['cp', 'cp_multiplier'], pokemon.pokemon_data):
+        return
+
+    candyToEvolve = int(pokeGameData.CandyToEvolve)
+
+    pokemon.candyNeededToMaxEvolve = pokemon_lvls[maxTCPM]['CandySoFar'] - pokemon_lvls[pokemon.cpm_total]['CandySoFar'] + candyToEvolve
+    pokemon.dustNeededToMaxEvolve = pokemon_lvls[maxTCPM]['DustSoFar'] - pokemon_lvls[pokemon.cpm_total]['DustSoFar']
+
+    i = 0
+    if pokemon.pokemon_id == 133: #is an Eevee
+        if pokemon.nickname is 'Sparky':
+            i = 2
+        elif pokemon.nickname is 'Pyro':
+            i = 3
+        else: #Rainer or Vaporean is the default
+            i = 1
+    else:
+        while game_master.get(pokemon.pokemon_id + i + 1, PokemonData()).FamilyId == pokeGameData.FamilyId and candyToEvolve > 0:
+            candyToEvolve = int(game_master.get(pokemon.pokemon_id + i + 1, PokemonData()).CandyToEvolve)
+            pokemon.candyNeededToMaxEvolve += candyToEvolve
+            i+=1
+
+    if(i == 0):
+        pokemon.maxEvolveCP = calcCP(pokemon.pokemon_data, maxTCPM, pokeGameData)
+    else:
+        evolvedPokeData = game_master.get(pokemon.pokemon_id + i , PokemonData())
+        pokemon.maxEvolveCP = calcCP(pokemon.pokemon_data, maxTCPM, evolvedPokeData)
+
+    pokeLvl = pokemon_lvls[pokemon.cpm_total]['PokemonLvl']
+    pokemon.PowerUpResult = calcCP(pokemon.pokemon_data, tcmpVals[pokeLvl], pokeGameData) - pokemon.cp
+
+#TCPM = CPM + ACPM
+#Stamina =  (BaseStamina + IndividualStamina) * TCPM
+#Attack = (BaseAttack + IndividualAttack) * TCPM
+#Defense = (BaseDefense + IndividualDefense) * TCPM
+#CP = MAX(10, FLOOR(Stamina0.5 * Attack * Def0.5 / 10))
+def calcACPM(pokemon, pokemon_details):
+    if not all_in(['cp', 'cp_multiplier'], pokemon.pokemon_data):
+        return 0    
+
+    baseAttk = int(pokemon_details.BaseAttack)
+    baseDef = int(pokemon_details.BaseDefense)
+    baseStamina = int(pokemon_details.BaseStamina)
+    cpm = pokemon.cp_multiplier
+    return max(0, sqrt(sqrt( (100*pow(pokemon.cp, 2)) / ( pow((baseAttk + pokemon.individual_attack), 2) * (baseDef + pokemon.individual_defense) * (baseStamina + pokemon.individual_stamina)  ) )) - cpm)
+
+def calcCP(pokemon, tcpm, pokemon_details):
+    baseAttk = int(pokemon_details.BaseAttack)
+    baseDef = int(pokemon_details.BaseDefense)
+    baseStamina = int(pokemon_details.BaseStamina)
+
+    attk = (baseAttk + pokemon.get('individual_attack', 0)) * tcpm
+    defense = (baseDef + pokemon.get('individual_defense', 0)) * tcpm
+    stamina = (baseStamina + pokemon.get('individual_stamina', 0)) * tcpm
+
+    return int(max(10, floor(sqrt(stamina) * attk * sqrt(defense) / 10)))
+
```

Comment:
i cross-checked and even though you multiply each individual value with tcpm while the other formula uses it inside the final formula as tcpm^2 it seems to even out :+1: